### PR TITLE
Remove game runner(#981)

### PR DESCRIPTION
**Pull request by RoiEXLab**
This adresses #981.
@DanVanAtta there's no need in updating install4J or gradle configurations they're all configured to run the GameRunner main method (which is still existing...)

---
This change is [Reviewable](https://reviewable.io/reviews/triplea-game/triplea/998)

### CODE REVIEW

File: src/games/strategy/engine/framework/GameRunner.java [hunk 1]
```diff
@@ -1,16 +1,108 @@
...
+import games.strategy.engine.framework.map.download.MapDownloadController;
+import games.strategy.engine.framework.startup.ui.MainFrame;
+import games.strategy.engine.lobby.server.LobbyServer;
+import games.strategy.triplea.ui.menubar.TripleAMenuBar;
+import games.strategy.ui.SwingAction;
+import games.strategy.util.CountDownLatchHandler;
+import games.strategy.util.EventThreadJOptionPane;
+import games.strategy.util.Version;
 
-/**
```
Comment (DanVanAtta) [inline src/games/strategy/engine/framework/GameRunner.java]: Please update rather than outright removing this javadoc

File: src/games/strategy/engine/framework/GameRunner.java [hunk 2]
```diff
@@ -1,16 +1,108 @@
...
+  public static final String TRIPLEA_NAME_PROPERTY = "triplea.name";
+  public static final String TRIPLEA_SERVER_PASSWORD_PROPERTY = "triplea.server.password";
+  public static final String TRIPLEA_STARTED = "triplea.started";
+  public static final String LOBBY_HOST = "triplea.lobby.host";
+  public static final String LOBBY_GAME_COMMENTS = "triplea.lobby.game.comments";
+  public static final String LOBBY_GAME_HOSTED_BY = "triplea.lobby.game.hostedBy";
+  public static final String LOBBY_GAME_SUPPORT_EMAIL = "triplea.lobby.game.supportEmail";
+  public static final String LOBBY_GAME_SUPPORT_PASSWORD = "triplea.lobby.game.supportPassword";
+  public static final String LOBBY_GAME_RECONNECTION = "triplea.lobby.game.reconnection";
+  // what is the default version of triplea (the one in the "bin" folder)
```
Comment (DanVanAtta) [inline src/games/strategy/engine/framework/GameRunner.java]: this comment is not valuable, IMO do not migrate it.

File: src/games/strategy/engine/framework/GameRunner.java [hunk 3]
```diff
@@ -1,16 +1,108 @@
...
- * <p>
- * This class is compiled to run under older jdks (1.3 at least), and should not do anything more than check the java
- * version
- * number, and then delegate to GameRunner2
- * <p>
- */
 public class GameRunner {
+
+  // not arguments:
+  public static final int PORT = 3300;
```
Comment (DanVanAtta) [inline src/games/strategy/engine/framework/GameRunner.java]: You've verified that all these class members still need to be public?

File: src/games/strategy/engine/framework/GameRunner.java [hunk 4]
```diff
@@ -1,16 +1,108 @@
...
+  public static final String TRIPLEA_STARTED = "triplea.started";
+  public static final String LOBBY_HOST = "triplea.lobby.host";
+  public static final String LOBBY_GAME_COMMENTS = "triplea.lobby.game.comments";
+  public static final String LOBBY_GAME_HOSTED_BY = "triplea.lobby.game.hostedBy";
+  public static final String LOBBY_GAME_SUPPORT_EMAIL = "triplea.lobby.game.supportEmail";
+  public static final String LOBBY_GAME_SUPPORT_PASSWORD = "triplea.lobby.game.supportPassword";
+  public static final String LOBBY_GAME_RECONNECTION = "triplea.lobby.game.reconnection";
+  // what is the default version of triplea (the one in the "bin" folder)
+  public static final String TRIPLEA_ENGINE_VERSION_BIN = "triplea.engine.version.bin";
+  // proxy stuff
```
Comment (DanVanAtta) [inline src/games/strategy/engine/framework/GameRunner.java]: Another comment that can be wiped out, it's obvious that PROXY_HOST and PROXY_PORT refer to 'proxy stuff'

File: src/games/strategy/engine/framework/headlessGameServer/HeadlessGameServer.java [hunk 5]
```diff
@@ -71,28 +71,28 @@
...
-        GameRunner2.TRIPLEA_SERVER_START_GAME_SYNC_WAIT_TIME, GameRunner2.TRIPLEA_SERVER_OBSERVER_JOIN_WAIT_TIME};
+    return new String[] {GameRunner.TRIPLEA_GAME_PROPERTY, TRIPLEA_GAME_HOST_CONSOLE_PROPERTY,
+        TRIPLEA_GAME_HOST_UI_PROPERTY, GameRunner.TRIPLEA_SERVER_PROPERTY, GameRunner.TRIPLEA_PORT_PROPERTY,
+        GameRunner.TRIPLEA_NAME_PROPERTY, GameRunner.LOBBY_HOST, LobbyServer.TRIPLEA_LOBBY_PORT_PROPERTY,
+        GameRunner.LOBBY_GAME_COMMENTS, GameRunner.LOBBY_GAME_HOSTED_BY, GameRunner.LOBBY_GAME_SUPPORT_EMAIL,
+        GameRunner.LOBBY_GAME_SUPPORT_PASSWORD, GameRunner.LOBBY_GAME_RECONNECTION,
+        GameRunner.TRIPLEA_SERVER_START_GAME_SYNC_WAIT_TIME, GameRunner.TRIPLEA_SERVER_OBSERVER_JOIN_WAIT_TIME};
   }
 
   private static void usage() {
```
Comment (DanVanAtta) [inline src/games/strategy/engine/framework/headlessGameServer/HeadlessGameServer.java]: Please move this usage function to below the first method that invokes it.

**Comment by DanVanAtta** [comment]
- Tests are not happy. Be sure you test this change out by hand as well, make sure the game launches as expected, and would be good to test that bots can launch as normal as well.

-  Couple of relatively minor comments

- GameRunner really needs to be cleaned up, but consolidation is a first step to get us there.

**Comment by RoiEXLab** [comment]
@DanVanAtta updated!

File: src/games/strategy/engine/ClientFileSystemHelper.java [hunk 6]
```diff
@@ -135,7 +132,7 @@ public static File getUserRootFolder() {
     final File userHome = new File(System.getProperties().getProperty("user.home"));
     // the default
     File rootDir;
-    if (GameRunner.isMac()) {
+    if (System.getProperties().getProperty("os.name").toLowerCase().contains("mac")) {
```
Comment (RoiEXLab) [inline src/games/strategy/engine/ClientFileSystemHelper.java]: @DanVanAtta Don't ask me why I can't use `GameRunner.isMac()` It keeps throwing Errors (not exceptions)

File: src/games/strategy/engine/ClientFileSystemHelper.java [hunk 7]
```diff
@@ -15,19 +9,22 @@
...
 public final class ClientFileSystemHelper {
 
-  private ClientFileSystemHelper() {
-
-  }
+  private ClientFileSystemHelper() {}
 
   public static File getRootFolder() {
-    final String fileName = getGameRunnerFileLocation("GameRunner2.class");
+    final String fileName = getGameRunnerFileLocation(GameRunner.class.getSimpleName() + ".class");
```
Comment (RoiEXLab) [inline src/games/strategy/engine/ClientFileSystemHelper.java]: Compiletime checks

File: src/games/strategy/engine/ClientFileSystemHelper.java (hunk 6 above)
Comment (DanVanAtta) [inline src/games/strategy/engine/ClientFileSystemHelper.java]: Can we accomplish the same logic without using system properties? Perhaps maybe just check if the mac directory is there first, if so use that, otherwise use the other value?

Do we use GameRunner.isMac() in other places, does it blow up there as well for you too? Could be a bit of a landmine waiting for you to execute it.

Comment (RoiEXLab) [inline src/games/strategy/engine/ClientFileSystemHelper.java]: The Problem is only occuring here... It seems that calling the 'GameRunner' class causes the error...
I think this has to do with the creation of client context, since in the javadoc something about calling gamerunner is mentioned - could be wrong...

TLDR; `GameRunner.isMac()` returns `System.getProperties().getProperty("os.name").toLowerCase().contains("mac")`, errors when calling from here.
Calling `System.getProperties().getProperty("os.name").toLowerCase().contains("mac")` directly works without problems

Comment (DanVanAtta) [inline src/games/strategy/engine/ClientFileSystemHelper.java]: That is the problem with ClientContext, since we are not using an IOC framework, whatever creates that context can't also be in turned called by client context.

Let's see if we can avoid "System.getProperties().getProperty("os.name").toLowerCase().contains("mac")". I do still think you could skip the check and just do:
```
if(  new File(new File(userHome, "Documents"), "triplea").exists() ) {
  // use it
} else { 
  rootDir = new File(userHome, "triplea");
 }
```

**Comment by RoiEXLab** [comment]
@DanVanAtta updated!
